src/url-params.ts

```diff
@@ -103,9 +103,18 @@ export class AnalyticsUrlParams {
 		if (this.client) {
 			throw new Error('Client is already set');
 		}
-		const newDeviceId = this.setDeviceIds(null, client.deviceId());
-		if (newDeviceId != null) {
-			client.setDeviceId(newDeviceId);
+		if (
+			this.getPassedDeviceId() &&
+			this.getPassedDeviceId() !== client.deviceId()
+		) {
+			const newDeviceId = this.setDeviceIds(null, client.deviceId());
+			if (newDeviceId != null) {
+				client.setDeviceId(newDeviceId);
+			}
+		}
+		if (this.sessionId && client.sessionId() !== this.sessionId) {
+			client.setSessionId(this.sessionId);
+			return;
```

Comment:
Why does the code `return` here? Any particular reason to skip the rest of this function?

I'd expect this to be `if .. else` statement instead and we call `this.client = client;` for both flows.

Reply:
Yeap that was an error on my side. I just fixed it. Thanks for noticing!

Reply:
Cool. @laziob, I don't see any new commit on this repo since my review.
(since you say "I just fixed it")

Reply:
I have it on local, was going to push once I get the other thing on the test.